### PR TITLE
TSK-1973: Check CREATED events to avoid null pointer

### PR DESCRIPTION
**Pull request by ArtemLtnr**
---
Release Notes:

```
Add check to avoid nullpointer
```
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] After integration of the PR, I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/jobs/HistoryCleanupJob.java [hunk 1]
```diff
@@ -161,17 +161,22 @@ private List<String> filterSameParentBusinessHistoryEventsQualifiedToClean(
...
     List<String> taskIdsToDeleteHistoryEventsFor = new ArrayList<>();
+    String createdKey = TaskHistoryEventType.CREATED.getName();
 
     taskHistoryIdsByEventTypeByParentBusinessProcessId.forEach(
         (parentBusinessProcessId, taskHistoryIdsByEventType) -> {
-          if (taskHistoryIdsByEventType.get(TaskHistoryEventType.CREATED.getName()).size()
+          if (!taskHistoryIdsByEventType.containsKey(createdKey)) {
+            LOGGER.error(
+                "Issue during history cleanup tasks with enabled parent business process. "
+                    + "No events for parent business process {} with type {} found.",
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/main/java/pro/taskana/simplehistory/impl/jobs/HistoryCleanupJob.java:171]:
```suggestion
                "Issue during history cleanup tasks with enabled parent business process. "
                    + "No events for parent business process {} with type {} found."
                    + "Please clean up those history events manually",
```

File: history/taskana-simplehistory-provider/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java [hunk 2]
```diff
@@ -374,6 +374,58 @@ void should_CleanHistoryEventsUntilDate_When_SameParentBusinessFalse() throws Ex
     assertThat(getHistoryService().createTaskHistoryQuery().count()).isEqualTo(16);
   }
 
+  @Test
+  @WithAccessId(user = "admin")
+  void should_NotCleanEvents_When_NoCreatedEvents_For_BusinessParentProcess() throws Exception {
```
Comment (mustaphazorgati) [inline history/taskana-simplehistory-provider/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java:379]:
```suggestion
  void should_NotCleanEvents_When_NoCreatedEventsForParentBusinessProcessIdExist() throws Exception {
```

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
looks very good! :)
some minor adjustments

File: history/taskana-simplehistory-provider/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java (hunk 2 above)
Comment (ArtemLtnr) [reply history/taskana-simplehistory-provider/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java]: done

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**